lab_3_generate_by_ngrams/main.py

```diff
@@ -221,6 +323,18 @@ def generate_next_token(self, sequence: tuple[int, ...]) -> Optional[dict]:
 
         In case of corrupt input arguments, None is returned
         """
+        if not isinstance(sequence, tuple) or\
+                len(sequence) == 0:
+            return None
+        next_tokens = {}
+        context = sequence[- self._n_gram_size + 1:]
+        if len(context) > len(sequence):
+            return None
+        for n_gram, freq in self._n_gram_frequencies.items():
+            if n_gram[:self._n_gram_size - 1] == context:
+                next_tokens[n_gram[:self._n_gram_size - 1]] = freq
+        sorted_tokens = dict(sorted(next_tokens.items(), key=lambda x: (x[1], -x[0][0], -x[0][1])))
```

Comment:
Sorting a dictionary is a bad idea. Do you know why?

Reply:
I tried to do it without sorting a dictionary and it started to pass tests